src/pages/library/DataBrowserDetails.js

```diff
@@ -190,7 +194,11 @@ const DataBrowserDetails = ({ id }) => {
             icon('angle-left', { size: 35 })
           ]),
           h(MainContent, { snapshot }),
-          h(Sidebar, { snapshot })
+          h(Sidebar, { snapshot, setShowRequestAccessModal }),
+          showRequestAccessModal && h(RequestDatasetAccessModal, {
+            datasets: [snapshot],
+            onDismiss: () => setShowRequestAccessModal(false)
+          })
```

Comment:
I would recommend moving this modal and the associated state to the `Sidebar` component since this is where the decision to show the modal is being made. This will also simplify the code a bit as the parent component will not need to be concerned about the modal state.